### PR TITLE
Added ruleset option to the preference page

### PR DESCRIPTION
**Pull request by rakshitkr**
- Automatic download of the rulesets from the nexus server based on the url.
- Update the rulesets on every eclipse restart.
- Integrated with preference page UI and the analysis.

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

### CODE REVIEW

File: plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java
Comment (kruegers) [inline plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java]: Please extract all URLS to an external config file.

File: plugins/de.cognicrypt.staticanalyzer/pom.xml
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/pom.xml]: Why do you need these dependencies here again? You've already defined them in the POM of core.

**Review by kruegers — CHANGES_REQUESTED** [review]

File: plugins/de.cognicrypt.staticanalyzer/pom.xml
Comment (rakshitkr) [reply plugins/de.cognicrypt.staticanalyzer/pom.xml]: You are right. These redundant dependencies have been removed.

File: plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java
Comment (rakshitkr) [reply plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java]: Do you mean a separate java file which consists of only URLs?

**Comment by sritejakv** [comment]
Before merging this branch, all the rule versions in the nexus repository should be changed from binary to source format. Now the plugin only supports rules in source format.

**Comment by sritejakv** [comment]
Before merging this branch, please merge [PR-188](https://github.com/CROSSINGTUD/CryptoAnalysis/pull/188) of CryptoAnalysis.

**Comment by sritejakv** [comment]
> Before merging this branch, all the rule versions in the nexus repository should be changed from binary to source format. Now the plugin only supports rules in source format.

As a solution for this, minimum rule version is added to skip all the rules in binary format.

File: plugins/de.cognicrypt.core/src/de/cognicrypt/utils/Utils.java
Comment (kruegers) [inline plugins/de.cognicrypt.core/src/de/cognicrypt/utils/Utils.java]: Please add proper exception handling.

File: plugins/de.cognicrypt.core/META-INF/MANIFEST.MF
Comment (kruegers) [inline plugins/de.cognicrypt.core/META-INF/MANIFEST.MF:35]: Why have you added this dependency now? This should not be required as all CrySL related classes must be made available through CryptoAnalysis.

File: plugins/de.cognicrypt.core/META-INF/MANIFEST.MF
Comment (kruegers) [inline plugins/de.cognicrypt.core/META-INF/MANIFEST.MF:54]: Same question as above for all  org.eclipse dependencies.

**Review by kruegers — CHANGES_REQUESTED** [review]
Sorry for the delayed review. I hadn't noticed you had addressed all my requests.
Also thanks for this work and for even implementing the customrules option. Please take care of the remaining three comments and the PR is ready to be merged. Thanks.

**Comment by kruegers** [comment]
Hey @rakshitkr,
The branch looks ok to me now, but unfortunately, you seem to have missed to [sign-off some of your commits](https://accounts.eclipse.org/legal/eca/validation/13425). Could you please take care of that? Then, the branch can be merged as is.

**Comment by kruegers** [comment]
Closed because replaced by #344 .

**Pull request closed without merging**
